core/src/components/datetime/datetime.tsx

```diff
@@ -657,7 +659,11 @@ export class Datetime implements ComponentInterface {
   }
 
   private initializeCalendarIOListeners = () => {
-    const { calendarBodyRef } = this;
+    /**
+     * Stencil sometimes sets calendarBodyRef to null on rerender, even though
+     * the element is present. Query for it manually as a fallback.
+     */
```

Comment:
I'm admittedly not sure why this happens 🤔 It's easiest to see if you log the `el` passed in when the `ref` is set in `renderCalendarBody`. I was able to consistently trigger it going between `date-time` and `time-date` presentations.

Reply:
This might just be a quirk in the timing of the `componentDidRender` vs. `componentDidLoad`. Although it is strange if that behavior is happening between subsequent renders.

Reply:
I thought that too, but I tried doing this:
```tsx
await new Promise(resolve => setTimeout(resolve, 2000));
console.log(this.calendarBodyRef)
```
and it was still `null`. You can also log the `el` in `renderCalendarBody` to see every time it updates; it'll fire twice, with the second one setting the ref to `null`.

Reply:
This looks like something that Stencil is doing under the hood when disposing a cached node. When the presentation changes, `renderDatetime` will return a different case and that must trigger behavior to tear down view refs before rendering the new case.

I think your fallback is appropriate considering this behavior (as long as it is the instance of the new `.calendar-body` and not the previously rendered instance).

Reply:
I'll grab some time with him to talk through it, and mark this as resolved for now 👍